### PR TITLE
[JENKINS-33358] Test against 2.73.x since that includes Groovy 2.4.11

### PR DESCRIPTION
**Pull request by jglick**
Like https://github.com/jenkinsci/workflow-cps-plugin/pull/175.

@reviewbybees

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -1 +1 @@
-buildPlugin(jenkinsVersions: [null, '2.32.3', '2.47'])
+buildPlugin(jenkinsVersions: [null, '2.32.3', '2.60.3', '2.73.1'])
```
Comment (jglick) [inline Jenkinsfile]: Probably you want to bump up the `jenkins.version` to 2.7.3, in which case we could drop the tests against 2.32.x I think.

File: Jenkinsfile [hunk 2]
```diff
@@ -1 +1 @@
-buildPlugin(jenkinsVersions: [null, '2.32.3', '2.47'])
+buildPlugin(jenkinsVersions: [null, '2.60.3', '2.73.1'])
```
Comment (jglick) [inline Jenkinsfile]: 2.7.x–2.47 uses one version of Groovy, 2.48–2.60.x another, 2.61+ yet another. At least we are dropping 1.x!

File: pom.xml [hunk 3]
```diff
@@ -62,8 +62,8 @@
         </pluginRepository>
     </pluginRepositories>
     <properties>
-        <jenkins.version>1.642.3</jenkins.version>
-        <workflow-cps-plugin.version>2.30</workflow-cps-plugin.version>
+        <jenkins.version>2.7.3</jenkins.version>
+        <workflow-cps-plugin.version>2.39</workflow-cps-plugin.version>
```
Comment (jglick) [inline pom.xml]: last security advisory

File: pom.xml [hunk 4]
```diff
@@ -108,7 +108,7 @@
         <dependency>
             <groupId>org.jenkins-ci.plugins</groupId>
             <artifactId>script-security</artifactId>
-            <version>1.27</version>
+            <version>1.33</version>
```
Comment (jglick) [inline pom.xml]: ditto, plus some follow-up fixes likely to affect testing of Groovy-heavy code like this

File: src/test/java/org/jenkinsci/plugins/workflow/cps/global/GrapeTest.java [hunk 5]
```diff
@@ -93,7 +93,6 @@
 
     // TODO test transitive dependencies; need to find something in Central which has a dependency not in this plugin’s test classpath and which could be used easily from a script
 
-    @Ignore("TODO ClassNodeResolver.tryAsLoaderClassOrScript swallows the CompilationFailedException and throws a generic GroovyBugError; fixed in Jenkins 2 / Groovy 2")
```
Comment (jglick) [inline src/test/java/org/jenkinsci/plugins/workflow/cps/global/GrapeTest.java]: Can finally have automated tests cover Jenkins 2 behaviors.

File: pom.xml (hunk 4 above)
Comment (abayer) [reply pom.xml]: We could theoretically go straight to 1.34, but 1.33 is good enough, yeah.

**Comment by jglick** [comment]
Webhooks broken pending https://github.com/kohsuke/github-api/pull/373, will try to retrigger.

**Comment by jglick** [comment]
@reviewbybees done

**Pull request merged**